cv_bridge/python/cv_bridge/core.py

```diff
@@ -132,6 +176,47 @@ def imgmsg_to_cv2(self, img_msg, desired_encoding = "passthrough"):
 
         return res
 
+    def cv2_to_compressed_imgmsg(self, cvim, dst_format = "jpg"):
+        """
+        Convert an OpenCV :cpp:type:`cv::Mat` type to a ROS sensor_msgs::CompressedImage message.
+
+        :param cvim:      An OpenCV :cpp:type:`cv::Mat`
+        :param dst_format:  The format of the image data, one of the following strings:
+
+           * from http://docs.opencv.org/2.4/modules/highgui/doc/reading_and_writing_images_and_video.html
+           * from http://docs.opencv.org/2.4/modules/highgui/doc/reading_and_writing_images_and_video.html#Mat imread(const string& filename, int flags)
+           * bmp, dib
+           * jpeg, jpg, jpe
+           * jp2
+           * png
+           * pbm, pgm, ppm
+           * sr, ras
+           * tiff, tif
+
+        :rtype:           A sensor_msgs.msg.CompressedImage message
+        :raises CvBridgeError: when the ``cvim`` has a type that is incompatible with ``format``
+
+
+        This function returns a sensor_msgs::Image message on success, or raises :exc:`cv_bridge.CvBridgeError` on failure.
+        """
+        import cv2, rospy, time
```

Comment:
rospy is a huge dependency, and just to convert a Python time.
Having now as a default does not make sense: now() is different from the time at which you acquired the image so it seems like a dangerous default.
The API could take a time as an argument (that would default to None and therefore (0,0)) but I would suggest this as a later CL as it would need to be added to the non-compressed API too.